scripts/servers-build.sh

```diff
@@ -17,11 +17,29 @@ build_servers_repositories() {
             cd go/src/github.com/mendersoftware/$repo
 
             case "$docker" in
-                generate-delta-worker|iot-manager|deployments|deployments-enterprise|deviceauth|deviceauth-enterprise|inventory|inventory-enterprise|tenantadm|useradm|useradm-enterprise|workflows|workflows-enterprise|create-artifact-worker|auditlogs|mtls-ambassador|deviceconnect|deviceconfig|devicemonitor|reporting)
+                iot-manager|deployments|deployments-enterprise|deviceauth|deviceauth-enterprise|inventory|inventory-enterprise|tenantadm|useradm|useradm-enterprise|workflows|workflows-enterprise|create-artifact-worker|auditlogs|mtls-ambassador|deviceconnect|deviceconfig|devicemonitor|reporting)
                     docker build -t $docker_url:pr .
                     $WORKSPACE/integration/extra/release_tool.py --set-version-of $docker --version pr
                     ;;
 
+                generate-delta-worker)
+                    local MENDER_BINARY_DELTA_VERSION=1.4.1
+                    local MENDER_ARTIFACT_VERSION=3.9.0
+                    local previous_id="${AWS_ACCESS_KEY_ID}"
+                    local previous_secret="${AWS_SECRET_ACCESS_KEY}"
+                    export AWS_ACCESS_KEY_ID="$AWSRO_MENDER_BINARY_DELTA_AWS_ACCESS_KEY_ID"
+                    export AWS_SECRET_ACCESS_KEY="$AWSRO_MENDER_BINARY_DELTA_AWS_SECRET_ACCESS_KEY"
+                    aws s3 cp s3://mender-binaries/mender-binary-delta/${MENDER_BINARY_DELTA_VERSION}/mender-binary-delta-${MENDER_BINARY_DELTA_VERSION}.tar.xz .
+                    xz -cd mender-binary-delta-${MENDER_BINARY_DELTA_VERSION}.tar.xz | tar xvf -;
+                    cp mender-binary-delta-${MENDER_BINARY_DELTA_VERSION}/x86_64/mender-binary-delta-generator .
+                    wget -q https://downloads.mender.io/mender-artifact/${MENDER_ARTIFACT_VERSION}/linux/mender-artifact -O mender-artifact
+                    docker build -t $docker_url:pr .
+                    $WORKSPACE/integration/extra/release_tool.py --set-version-of $docker --version pr
+                    export AWS_ACCESS_KEY_ID="$previous_id"
+                    export AWS_SECRET_ACCESS_KEY="$previous_secret"
+                    rm -Rf "mender-binary-delta-${MENDER_BINARY_DELTA_VERSION}"
+                    ;;
```

Comment:
The build steps seem quite complex and difficult to maintain.  I also assume we have the same steps in the individual pipeline of this repository. It is difficult to maintain in the sense that `mender-qa` (un-versioned repository) needs to closely follow changes in `generate-delta-workder` (versioned).

I therefore suggest moving `generate-delta-worker` out of the `servers-build.sh` and include the build job from the individual pepline.

To be able to do that the individual pipeline needs to have two different gitlab yaml files: one with only the build (which can be included from here) and one with everything else (which also includes the local one for the build). See [`mender-gateway` pipeline design](https://github.com/mendersoftware/mender-gateway/blob/master/.gitlab-ci.yml#L17) for inspiration